bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/model/RouterosDevice.java

```diff
@@ -0,0 +1,278 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mikrotik.internal.model;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.net.SocketFactory;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import me.legrange.mikrotik.ApiConnection;
+import me.legrange.mikrotik.ApiConnectionException;
+import me.legrange.mikrotik.MikrotikApiException;
+
+/**
+ * The {@link RouterosDevice} class is wrapped inside a bridge thing and responsible for communication with
+ * Mikrotik device, data fetching, caching and aggregation.
+ *
+ * @author Oleg Vivtash - Initial contribution
+ */
+@NonNullByDefault
+public class RouterosDevice {
+    private final Logger logger = LoggerFactory.getLogger(RouterosDevice.class);
+
+    private final String host;
+    private final int port;
+    private final int connectionTimeout;
+    private final String login;
+    private final String password;
+    @Nullable
+    private ApiConnection connection;
+
+    public static final String PROP_ID_KEY = ".id";
+    public static final String PROP_TYPE_KEY = "type";
+    public static final String PROP_NAME_KEY = "name";
+    public static final String PROP_SSID_KEY = "ssid";
+
+    private static final String CMD_PRINT_IFACES = "/interface/print";
+    private static final String CMD_PRINT_IFACE_TYPE_TPL = "/interface/%s/print";
+    private static final String CMD_MONTOR_IFACE_MONITOR_TPL = "/interface/%s/monitor numbers=%s once";
+    private static final String CMD_PRINT_CAPS_IFACES = "/caps-man/interface/print";
+    private static final String CMD_PRINT_CAPSMAN_REGS = "/caps-man/registration-table/print";
+    private static final String CMD_PRINT_WIRELESS_REGS = "/interface/wireless/registration-table/print";
+    private static final String CMD_PRINT_RESOURCE = "/system/resource/print";
+    private static final String CMD_PRINT_RB_INFO = "/system/routerboard/print";
+
+    @Nullable
+    private List<RouterosInterfaceBase> interfaceCache;
+    @Nullable
+    private List<RouterosCapsmanRegistration> capsmanRegistrationCache;
+    @Nullable
+    private List<RouterosWirelessRegistration> wirelessRegistrationCache;
+    private Set<String> monitoredInterfaces = new HashSet<>();
+    private Map<String, String> wlanSsid = new HashMap<>();
+
+    @Nullable
+    private RouterosSystemResources resourcesCache;
+    @Nullable
+    private RouterosRouterboardInfo rbInfo;
+
+    private static Optional<RouterosInterfaceBase> createTypedInterface(Map<String, String> interfaceProps) {
+        @Nullable
+        RouterosInterfaceType ifaceType = RouterosInterfaceType.resolve(interfaceProps.get(PROP_TYPE_KEY));
```

Comment:
The compiler is more intelligent on local variables. Therefore, the Nullable could be removed. Please check all.

Reply:
Done